internal/server/data/migrations.go

```diff
@@ -1329,3 +1330,52 @@ func addAccessKeyIssuedForKind() *migrator.Migration {
 		},
 	}
 }
+
+func storeProviderUserGroupsArray() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2023-01-26T13:37",
+		Migrate: func(tx migrator.DB) error {
+			var groupType string
+			if err := tx.QueryRow(`SELECT pg_typeof(groups) FROM provider_users LIMIT 1;`).Scan(&groupType); err != nil {
+				return err
+			}
+			if groupType == "jsonb" {
+				// this migration has already been performed
+				return nil
+			}
+			if _, err := tx.Exec(`ALTER TABLE provider_users ADD COLUMN IF NOT EXISTS groups_json jsonb;`); err != nil {
```

Comment:
Does this need a temporary column? It seems you can retrieve the user groups, change the column, then reinsert the rows, e.g. add ALTER TABLE between the scanRows on line 1359 and the for loop on line 1366

Reply:
Good observation, made this change.